### ISSUE TITLE
Constantly getting: failed to write batch: [403] 403 Forbidden error while using Azure monitor as output

### ISSUE THREAD

**Issue by kishorebasisth** [influxdata/telegraf#6277]
### Relevant telegraf.conf:

```
[agent]
 interval = "10s"
 round_interval = true
 metric_buffer_limit = 1000
 flush_buffer_when_full = true
 flush_interval = "5s"
 quiet = false
 logfile = ""
 ## Override default hostname, if empty use os.Hostname()
 hostname = "test-vm"

[[outputs.azure_monitor]]
 resource_id = "/subscriptions/XXXXXXXXXX/resourcegroups/XXXXXXXXXXXX/providers/microsoft.operationalinsights/workspaces/XXXXXXXXXXXXX"
 strings_as_dimensions = true
 region = "westeurope"

[[inputs.win_perf_counters]]
 [[inputs.win_perf_counters.object]]
  ObjectName = "Processor"
  Instances = ["*"]
  Counters = ["% Idle Time", "% Interrupt Time", "% Privileged Time", "% User Time", "% Processor Time"]
  Measurement = "win_cpu"
  IncludeTotal=true
 [[inputs.win_perf_counters.object]]
  ObjectName = "LogicalDisk"
  Instances = ["*"]
  Counters = ["Free Megabytes", "Disk Reads/sec", "Disk Writes/sec", "Disk Bytes/sec", "% Idle Time", "Current Disk Queue Length"]
  Measurement = "win_disk"
 [[inputs.win_perf_counters.object]]
  ObjectName = "Network Interface"
  Counters = ["Bytes Sent/sec", "Bytes Received/sec"]
  Instances = ["*"]
  Measurement = "win_network"
 [[inputs.win_perf_counters.object]]
  ObjectName = "Memory"
  Counters = ["Available Bytes", "Committed Bytes", "Cache Bytes", "Pages/sec", "Pool Nonpaged Bytes", "Pool Paged Bytes"]
  Instances = ["------"]
  Measurement = "win_mem"

```

### System info:

.\telegraf.exe --version
Telegraf 1.11.4 (git: HEAD d9ca76e4)

Windows Server 2016 and Windows 10

### Steps to reproduce:

1. Create the following environment variable using powershell:
```
setx AZURE_TENANT_ID "xxxxxxxxxxxx"
setx AZURE_CLIENT_ID "xxxxxxxxxxxxxxxxx"
setx AZURE_CLIENT_SECRET "xxxxxxxxxxxx"
```
2. Install telegraf version "Telegraf 1.11.4"
3. Copy the config file and run telegraf

### Expected behavior:
Metrics being sent to Azure monitor
### Actual behavior:
Following error occurs:
```
2019-08-16T13:38:31Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"test-vm", Flush Interval:
2019-08-16T13:39:02Z E! [agent] Error writing to output [azure_monitor]: failed to write batch: [403] 403 Forbidden
2019-08-16T13:39:05Z E! [agent] Error writing to output [azure_monitor]: failed to write batch: [403] 403 Forbidden
2019-08-16T13:39:10Z E! [agent] Error writing to output [azure_monitor]: failed to write batch: [403] 403 Forbidden
2019-08-16T13:39:15Z E! [agent] Error writing to output [azure_monitor]: failed to write batch: [403] 403 Forbidden
2019-08-16T13:39:20Z E! [agent] Error writing to output [azure_monitor]: failed to write batch: [403] 403 Forbidden
2019-08-16T13:39:25Z E! [agent] Error writing to output [azure_monitor]: failed to write batch: [403] 403 Forbidden
2019-08-16T13:39:30Z E! [agent] Error writing to output [azure_monitor]: failed to write batch: [403] 403 Forbidden
```
### Additional info:

The azure credentials used have access to Azure subscription as contributor. No matter which service prinicpal details I try, I always get the aforementioned error. Even on my local windows 10 computer I get this error.

**Comment by danielnelson** [comment]
I'm not sure how to fix, but it looks like your credentials are valid but this client is not allowed to write to the resource.  Try removing the `resource_id` so that Telegraf will pick it automatically?

**Comment by kishorebasisth** [comment]
I get the same error even after removing the resource id tag. The service principal details used have contributor access to the whole subscription. I find it strange that a client that has contributor access cannot write to a resource.

**Comment by dbaltor** [comment]
I'm getting the same error message even using Azure "system-assigned managed identity" to authenticate the agent (https://docs.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/qs-configure-portal-windows-vm#enable-system-assigned-managed-identity-on-an-existing-vm). To add insult to injure, the same setup in another ubuntu VM is working just fine.

**Comment by dbaltor** [comment]
I tried to use the same configuration mentioned by @kishorebasisth, **Client Credentials: Azure AD Application ID and Secret**,  as described [here](https://github.com/influxdata/telegraf/tree/master/plugins/outputs/azure_monitor#authentication) but ended up getting the same 403 error on the logs.
Not sure, but I suspect it is due to the step 2:
> A Service Principal or User Principal needs to be assigned the Monitoring Metrics Publisher role on the resource(s) metrics will be emitted against.

**Comment by dbaltor** [comment]
Things have evolved and actually I'm getting the following error in the logs:
`2019-11-05T15:57:50Z E! [agent] Error writing to outputs.azure_monitor: unable to fetch authentication credentials: azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://uksouth.monitoring.azure.com/subscriptions/<subscription-id>/resourceGroups/<resource-group>/providers/Microsoft.Compute/virtualMachines/azuredemo-ops-manager-vm/metrics: StatusCode=400 -- Original Error: adal: Refresh request failed. Status Code = '400'. Response body: {"error":"invalid_request","error_description":"Identity not found"}`
which seems to be a problem with the Azure authentication package.
I've opened an issue on their repo:
[#482](https://github.com/Azure/go-autorest/issues/482)

**Comment by dbaltor** [comment]
We managed to recompile Telegraf agent using the latest version of the Azure/go-autorest package and this solved the issue, in other words, the new binary is able to authenticate against AAD using the env vars.
I've opened the PR #6656

**Comment by Slowdive-Aideron** [comment]
Hey, I'm curious why you put the workspace "/subscriptions/XXXXXXXXXX/resourcegroups/XXXXXXXXXXXX/providers/microsoft.operationalinsights/workspaces/XXXXXXXXXXXXX"
 in the resource_id ... I thought that has to be the VM from where the metrics are comming from

**Comment by sjwang90** [comment]
Fixed #6656
